crates/rattler/src/install/clobber_registry.rs

```diff
@@ -0,0 +1,648 @@
+
+        let mut temp_clobbers = Vec::new();
+        for prefix_record in prefix_records {
+            registry
+                .package_names
+                .push(prefix_record.repodata_record.package_record.name.clone());
+
+            let clobber_ending =
+                clobber_template(&prefix_record.repodata_record.package_record.name);
+
+            for p in &prefix_record.files {
+                if p.to_string_lossy().ends_with(&clobber_ending) {
+                    // register a clobbered path
+                    if let Some(full_file_name) = p.file_name() {
+                        if let Some((file_name, originating_package)) = full_file_name
+                            .to_string_lossy()
+                            .split_once(CLOBBER_TEMPLATE)
+                        {
+                            let path = p.with_file_name(file_name);
+                            temp_clobbers.push((path, originating_package.to_string()));
+                        }
+                    }
+                } else {
+                    registry
+                        .paths_registry
+                        .insert(p.clone(), registry.package_names.len() - 1);
+                }
+            }
+        }
+
+        for (path, originating_package) in temp_clobbers.iter() {
+            let idx = registry
+                .package_names
+                .iter()
+                .position(|n| n.as_normalized() == *originating_package)
+                .unwrap();
+
+            registry
+                .clobbers
+                .entry(path.clone())
+                .or_insert_with(|| vec![*registry.paths_registry.get(path).unwrap()])
+                .push(idx);
+        }
+
+        registry
+    }
+
+    fn clobber_name(path: &Path, package_name: &PackageName) -> PathBuf {
+        let file_name = path.file_name().unwrap_or_default();
+        let mut new_path = path.to_path_buf();
+        new_path.set_file_name(format!(
+            "{}{}",
+            file_name.to_string_lossy(),
+            clobber_template(package_name),
+        ));
+        new_path
+    }
+
+    /// Register the paths of a package when linking the package.
+    pub fn register_paths(
+        &mut self,
+        name: &str,
+        paths_json: &PathsJson,
+    ) -> HashMap<PathBuf, PathBuf> {
+        let mut clobber_paths = HashMap::new();
+
+        // check if we have the package name already registered
+        let name_idx = if let Some(idx) = self
+            .package_names
+            .iter()
+            .position(|n| n.as_normalized() == name)
+        {
+            idx
+        } else {
+            self.package_names
+                .push(PackageName::try_from(name).unwrap());
+            self.package_names.len() - 1
+        };
+
+        for entry in paths_json.paths.iter() {
+            let path = entry.relative_path.clone();
+
+            if let Entry::Vacant(e) = self.paths_registry.entry(path.clone()) {
```

Comment:
I think it would be better to use `get` here instead of `entry`. So you dont have to clone the `path`. Only if no entry exists yet.